Engine/engine.cpp

```diff
@@ -501,31 +505,48 @@ void Engine::_encodeColumnNamesinOptions(Json::Value & options, Json::Value & me
 	if(meta.isNull())
 		return;
 	
-	bool encodePlease = meta.isObject() && meta.isMember("containsColumn") && meta["containsColumn"].asBool();
+	bool	encodePlease	= meta.isObject() && meta.get("shouldEncode",	false).asBool(),
+			isRCode			= meta.isObject() && meta.get("rCode",			false).asBool();
 
 	switch(options.type())
 	{
 	case Json::arrayValue:
 		if(encodePlease)
 			ColumnEncoder::columnEncoder()->encodeJson(options, false); //If we already think we have columnNames just change it all
+		
 		else if(meta.type() == Json::arrayValue)
 			for(size_t i=0; i<options.size() && i < meta.size(); i++)
 				_encodeColumnNamesinOptions(options[i], meta[i]);
-		else if(meta.type() == Json::objectValue && meta.get("containsColumn", false).asBool())
+		
+		else if(isRCode)
+		{
+			for(size_t i=0; i<options.size(); i++)
+				if(options[i].isString())
+					options[i] = ColumnEncoder::columnEncoder()->encodeRScript(options[i].asString());
+		}
+		else if(encodePlease)
```

Comment:
This extra `else if(encodePlease)` is quite redundant.

Reply:
Indeed indeed, ive fixed it now. Not sure how I missed that before but ah well :p